lib/waterfoul/ppu.rb

```diff
@@ -37,19 +37,19 @@ def step(cycles = 1)
       @modeclock += cycles
       @auxillary_modeclock += cycles
 
-      if IO::LCDControl.screen_enabled? && @screen_enabled
-        case @mode
-        when H_BLANK_STATE
-          hblank if @modeclock >= H_BLANK_TIME
-        when V_BLANK_STATE
-          vblank
-        when OAM_READ_STATE
-          oam if @modeclock >= OAM_SCANLINE_TIME
-        when VMRAM_READ_STATE
-          vram if @modeclock >= VRAM_SCANLINE_TIME
-        end
-      else
-        if IO::LCDControl.screen_enabled?
+      if IO::LCDControl.screen_enabled?
```

Comment:
Compare https://github.com/eregon/waterfoul/blob/f095a087a8e9d1f64eb105b2e4caff4b2d4e4b8b/lib/waterfoul/ppu.rb#L40-L64 and https://github.com/eregon/waterfoul/blob/18e62d17221644a79aa88b3b7f9952ea8803a75d/lib/waterfoul/ppu.rb#L40-L64